src/build/app/mod.rs

```diff
@@ -1481,40 +1470,160 @@ impl<'b> App<'b> {
                 .collect();
 
             if !args_missing_help.is_empty() {
-                let abort_message = format!("AppSettings::HelpRequired is enabled for the App {}, but at least one of its arguments does not have either `help` or `long_help` set. List of such arguments: {}", app.name, args_missing_help.join(", "));
-                panic!(abort_message);
+                panic!(format!(
+                    "AppSettings::HelpRequired is enabled for the App {}, but at least one of its arguments does not have either `help` or `long_help` set. List of such arguments: {}",
+                    self.name,
+                    args_missing_help.join(", ")
+                ));
             }
         }
-        for sub_app in &app.subcommands {
-            Self::_panic_on_missing_help(&sub_app, help_required_globally);
+
+        for sub_app in &self.subcommands {
+            sub_app._panic_on_missing_help(help_required_globally);
         }
     }
 
     // Perform some expensive assertions on the Parser itself
-    fn _app_debug_asserts(&self) -> bool {
-        debugln!("App::_app_debug_asserts;");
-        for name in self.args.args.iter().map(|x| x.id) {
-            if self.args.args.iter().filter(|x| x.id == name).count() > 1 {
-                panic!(format!(
-                    "Arg names must be unique, found '{}' more than once",
-                    name
-                ));
+    #[allow(clippy::cognitive_complexity)]
+    fn _debug_asserts(&self) -> bool {
+        debugln!("App::_debug_asserts;");
+
+        for arg in &self.args.args {
+            debug_assert!(arg._debug_asserts());
+
+            // Name conflicts
+            assert!(
+                self.args.args.iter().filter(|x| x.id == arg.id).count() < 2,
+                "Argument name must be unique\n\n\t'{}' is already in use",
+                arg.name,
+            );
+
+            // Long conflicts
+            if let Some(l) = arg.long {
+                assert!(
+                    self.args.args.iter().filter(|x| x.long == Some(l)).count() < 2,
+                    "Argument long must be unique\n\n\t'--{}' is already in use",
+                    l
+                );
+            }
+
+            // Short conflicts
+            if let Some(s) = arg.short {
+                assert!(
+                    self.args.args.iter().filter(|x| x.short == Some(s)).count() < 2,
+                    "Argument short must be unique\n\n\t'-{}' is already in use",
+                    s
+                );
+            }
+
+            // Index conflicts
+            if let Some(idx) = arg.index {
+                assert!(
+                    positionals!(self).fold(0, |acc, p| if p.index == Some(idx) {
+                        acc + 1
+                    } else {
+                        acc
+                    }) < 2,
+                    "Argument '{}' has the same index as another positional argument\n\n\t \
```

Comment:
Would be nice to print the name of the other arg here

Reply:
I haven't touched this at all and don't want to yet because we might change the `positionals!` macro.